numpy/lib/nanfunctions.py

```diff
@@ -395,11 +415,12 @@ def nanmax(a, axis=None, out=None, keepdims=np._NoValue):
         if mask is None:
             return res
 
-        # Check for all-NaN axis
-        mask = np.all(mask, axis=axis, **kwargs)
-        if np.any(mask):
-            res = _copyto(res, np.nan, mask)
-            warnings.warn("All-NaN axis encountered", RuntimeWarning, stacklevel=2)
+        if initializer is np._NoValue:
+            # Check for all-NaN axis
+            mask = np.all(mask, axis=axis, **kwargs)
+            if np.any(mask):
+                res = _copyto(res, np.nan, mask)
+                warnings.warn("All-NaN axis encountered", RuntimeWarning, stacklevel=2)
```

Comment:
What if `initializer` is NaN?

Reply:
Hmm. I was unsure what to do there, actually. On one hand, it may mean that the user WANTS `nan` values in the output. On the other, it would give `nan`s.

Reply:
Fine as is for now I think.

Reply:
Works for me

Reply:
I fixed the potential issue. Wasn't too hard. :-)